### ISSUE TITLE
Http Integration Tests: Create tests to support HTTP

### ISSUE THREAD

**Issue by pratikpandey21** [DiceDB/dice#734]
This issue involves creating the below test cases for HTTP as well:

**Source**: integration_tests/commands/async/

- bit_operation_test.go
- bit_ops_string_int_test.go
- check_type_test.go
- command_count_test.go
- command_default_test.go

Steps:
1) Create the corresponding tests for these commands in integration_tests/commands/http.
2) Use set_test.go as a reference, and how to alter the input and output to match JSON structures from RESP.
3) Run the tests and fix issues with command parsing in HTTP.

If you need to do anything **custom** to support HTTP, let @pratikpandey21 @lucifercr07 know.

**Comment by svkaizoku** [comment]
Hey @pratikpandey21 , I would like to pick this up. Can you add the tag if this will be a good first issue? I am new to `go`, but would like to take dig at this issue. wdyt?

**Comment by lucifercr07** [comment]
@svkaizoku assigned, thanks for contributing. This issue should be good for you to start with.

**Comment by svkaizoku** [comment]
Thanks @lucifercr07 , will start work on it over the weekend :)

**Comment by arpitbbhayani** [comment]
Hello @svkaizoku,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by svkaizoku** [comment]
Hey @arpitbbhayani , I am actively working on this. I am creating separate PR's for these, that are referenced in this issue as as you can see above. Currently I am not stuck in anything. Will reach out in discord if any. Thanks :D

**Comment by svkaizoku** [comment]
So just in case, Here is the update

[#734: Http Integration tests for COMMAND and COMMAND/COUNT](https://github.com/DiceDB/dice/pull/899) : In Review 
[#734 : Http Commands Error Messages Test #900](https://github.com/DiceDB/dice/pull/900) : In Review 
[Http Integration Tests for Bit Operations](https://github.com/DiceDB/dice/pull/870) Working on it.

**Comment by svkaizoku** [comment]
Hey @lucifercr07 , I am unabe to write proper tests for `bit_operation_test.go`,`bit_ops_string_int_test.go`
due to this issue https://github.com/DiceDB/dice/issues/942 . 
As you can see from this [Http Integration Tests for Bit Operations](https://github.com/DiceDB/dice/pull/870/files) , I have covered half of the working cases, but there seems to be issues with encoding. Once the issue is fixed, it would make sense to write proper tests for it.

**Comment by svkaizoku** [comment]
@JyotinderSingh can we reopen it. The whole task is not completed yet as tests for 

> Hey @lucifercr07 , I am unabe to write proper tests for `bit_operation_test.go`,`bit_ops_string_int_test.go` due to this issue #942 . As you can see from this [Http Integration Tests for Bit Operations](https://github.com/DiceDB/dice/pull/870/files) , I have covered half of the working cases, but there seems to be issues with encoding. Once the issue is fixed, it would make sense to write proper tests for it.

@JyotinderSingh can we reopen this issue as it is not completed due to it being blocked by the above mentioned issue.
